Web_Development/FrontEnd/HTML/html_conventions_practices.md

```diff
@@ -0,0 +1,408 @@
+    </body>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <BODY>
+        <p>Winter of Contributing</p>
+    </BODY>
+    ```
+
+### Use Lowercase Attribute Names
+
+HTML allows the use of lowercase or uppercase element names; we recommend you to opt for lowercase names, because:
+
+-   It looks cleaner
+-   Easier to write and read
+-   Mixing uppercase and lowercase names looks bad
+
+-   Good Practice:
+
+    ```sh
+    <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px">
+    <a href="https://gwoc.girlscript.tech/">Visit GWOC</a>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <img SRC="gwoc.gif" ALT="GWOC" STYLE="width:128px;height:128px">
+    <a HREF="https://gwoc.girlscript.tech/">Visit GWOC</a>
+    ```
+
+### Indentation in HTML
+
+It is not recommended to use indentation unnecessarily in an HTML document. Indentation should be limited to two spaces rather than using the tab key(Tab key's space is not a valid space).
+
+-   Good Practice:
+
+    ```sh
+    <body>
+      <h1>Open Source</h1>
+
+      <h2>GSSOC</h2>
+      <p>It will happen in March</p>
+      <h2>GWOC</h2>
+      <p>It will happen in September</p>
+    </body>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <body>
+      <h1>Open Source</h1>
+      <h2>GSSOC</h2>
+        <p>
+          It will happen in March
+        </p>
+      <h2>GWOC</h2>
+        <p>
+          It will happen in September
+        </p>
+    </body>
+    ```
+
+### Always Quote Attribute Values
+
+HTML allows attribute values without quotes. However, we recommend quoting attribute values, because:
+
+-   Quoted values are easier to read
+-   One must also ensure not to use spaces
+
+-   Good Practice:
+
+    ```sh
+    <body class="container">
+      <p>Winter of Contributing</p>
+    </body>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <body class=container>
+      <p>Winter of Contributing</p>
+    </body>
+    ```
+
+    ```sh
+    <body class=body container>
+      <p>Winter of Contributing</p>
+    </body>
+    ```
+
+### Close All HTML Elements
+
+In HTML, Sometimes if you run without closing any tag, It won't give you a syntax error.
+
+Well, If the code/element is working without closing tag, then that doesn't make sense or a good practice to not to close the tags.
+
+-   Good Practice:
+
+    ```sh
+    <body>
+      <p>Girlscript Foundation</p>
+      <p>Winter of Contributing</p>
+    </body>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <body>
+      <p>Girlscript Foundation
+      <p>Winter of Contributing
+    </body>
+    ```
+
+### Always Specify required Attributes
+
+Like Images,
+
+Always specify the alt attribute for images. This attribute is important if the image for some reason cannot be displayed.
+
+Also, always define the width and height of images.Because the browser can reserve space for the image before loading and sometimes image ratio differs.
+
+-   Good Practice:
+
+    ```sh
+    <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px"/>
```

Comment:
Avoid suggesting the use of inline styles.

Reply:
yeah